miditok/midi_tokenizer.py

```diff
@@ -1663,30 +1663,32 @@ def apply_bpe_to_dataset(
     ):
         r"""Applies BPE to an already tokenized dataset (with no BPE).
 
-        :param dataset_path: path to token files to load.
+        :param dataset_path: path to the directory containing token json files.
         :param out_path: output directory to save. If none is given, this method will overwrite original files.
                 (default: None)
         """
         if not self.has_bpe:
             return
+        if isinstance(out_path, str):
+            out_path = Path(out_path)
 
         files_paths = list(Path(dataset_path).glob("**/*.json"))
-        for path in tqdm(files_paths, desc="Applying BPE to dataset"):
-            sample = self.load_tokens(path)
+        for json_path in tqdm(files_paths, desc="Applying BPE to dataset"):
+            sample = self.load_tokens(json_path)
             seq = (
                 TokSequence(ids=sample["ids"])
                 if self.one_token_stream
                 else [TokSequence(ids=track) for track in sample["ids"]]
             )
             self.apply_bpe(seq)
 
-            out_ = (
-                Path(out_path) / path.relative_to(dataset_path)
+            saving_path = (
+                out_path / json_path.relative_to(dataset_path)
                 if out_path is not None
-                else path
+                else json_path
             )
```

Comment:
After that line, I think there is a `saving_path.parent.mkdir(parents=True, exist_ok=True)` missing as I will raise an error if the directory structure does not exist before writing the file.

Reply:
```suggestion
            )
            saving_path.parent.mkdir(parents=True, exist_ok=True)
```